### ISSUE TITLE
Entry property TextTransform = TextTransform.None does not work on iOS

### ISSUE THREAD

**Issue by czuck** [dotnet/maui#21024]
### Description

iOS will upper-case the first letter of an entry by default.  According to the documentation, setting the TextTransformation to None indicates that the text will not be transformed, but the first letter is still being capitalized.

### Steps to Reproduce

1) add an Entry to a page and set the TextTransform to none such as this:

```xaml
                <Entry x:Name="Entry"
                                         WidthRequest="150"
                                         Text=""
                                         TextTransform="None"
                                         VerticalOptions="Center"/>
```
2) Run the app, navigate to the page, and type into the entry in all lower case.  The first letter will be uppercase.

### Version with bug

8.0.7 SR2

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

iOS

### Affected platform versions

iOS 17.2

### Did you find any workaround?

Create a custom handler and set  platformView.AutocapitalizationType = UITextAutocapitalizationType.None;

**Comment by jfversluis** [comment]
I think the assumption about what `TextTransform` in this case does might be wrong. According to [the docs](https://learn.microsoft.com/dotnet/maui/user-interface/controls/entry?view=net-maui-8.0#transform-text):

> An [Entry](https://learn.microsoft.com/dotnet/api/microsoft.maui.controls.entry) can transform the casing of its text, stored in the Text property, by setting the TextTransform property to a value of the TextTransform enumeration.

This will only be applied to the backing field, not the keyboard or suggested corrections. There is currently not built-in way for .NET MAUI to influence the behavior besides implementing a customer handler as you have done.

**Comment by czuck** [comment]
@jfversluis I'm not sure I understand.  If I set TextTransform to Uppercase and enter text in all lowercase what I see in the entry on the device is all uppercase.  If I set TextTransform to Lowercase and type exactly the same the text in the entry is all lowercase.  If I leave TextTransform as Default and type I get the first letter capitalized, which is the iOS default.  What I want is for the text not to be changed at all and appear exactly as I have typed, which is what the documentation says setting TextTransform to None does.
